avro-serializer/src/main/java/io/confluent/kafka/serializers/GenericContainerWithVersion.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.serializers;
+
+import org.apache.avro.generic.GenericContainer;
+
+import java.util.Objects;
+
+/**
+ * Wrapper for GenericContainer along with a version number.
```

Comment:
We probably need to be much more explicit about whether a version is required, when it is expected, and what the behavior is when the version is null. If it is not expected to be null, then we should consider using `int` rather than `Integer`, or add asserts or checks accordingly.

Reply:
Yes, I'll add a comment that it can be null.